adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -119,6 +118,480 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
   def transform(tFn: RDD[T] => RDD[T]): U = {
     replaceRdd(tFn(rdd))
   }
+  val self = this
+
+  /**
+   *
+   */
+  private[rdd] object SortedTrait extends Serializable {
+    var isSorted: Boolean = false
+    var partitionMapRdd: RDD[(ReferenceRegion, ReferenceRegion)] = null
+    lazy val partitionMap: Seq[(ReferenceRegion, ReferenceRegion)] = partitionMapRdd.collect
+
+    def apply(newRdd: RDD[T], newPartitionMapRDD: RDD[(ReferenceRegion, ReferenceRegion)]): U = {
+      val newObject = replaceRdd(newRdd)
+      newObject.SortedTrait.isSorted = true
+      newObject.SortedTrait.partitionMapRdd = newPartitionMapRDD
+      newObject
+    }
+
+    /**
+     * This repartition method repartitions all data in this.rdd and distributes it as evenly as possible
+     * into the number of partitions provided. The data in this.rdd is already sorted (by ReferenceRegion.start)
+     * so we ensure that it is kept that way as we move data around.
+     *
+     * Data is repartitioned in lists to maintain the sorted order. After the repartition, the lists are placed
+     * in the correct order based on their original sorted order.
+     *
+     * @param partitions the number of partitions to repartition this.rdd into
+     * @return a new SortedGenomicRDDMixIn with the RDD partitioned evenly
+     */
+    private def evenlyRepartition(partitions: Int)(implicit tTag: ClassTag[T]): GenomicRDD[T, U] = {
+      if (!isSorted) return self
+      val collectedPartitionMap = partitionMap
```

Comment:
Rename? This line is required to ensure that the partitionMap is materialized before the executors try to access it.